isobus/src/isobus_virtual_terminal_server.cpp

```diff
@@ -171,11 +171,11 @@ namespace isobus
 				CANStackLogger::debug("[VT Server]: Executing macro %u", macro->get_id());
 				retVal = true;
 
-				for (std::uint16_t j = 0; j < macro->get_number_of_commands(); j++)
```

Comment:
I actually had this as uint16 to avoid an infinite loop in the case where exactly 255 macros were included

Reply:
Hmm, that's strange. The `get_number_of_macros` function returns an `uint8`, so a loop of `uint16` doesnt make sense to me as a fix for that issue

Reply:
255 commands, rather. Like if get_number_of_commands returns 255, our counter here would roll over

Reply:
Sorry I meant get_number_of_commands indeed. That is still a uint8_t, so max would be 255 commands as you said. 

Here we are speaking about a **strictly less** loop, which would mean the loop stops at `i=254` since `(i=255) < (255=max)` doesn't hold (which is also the index of the command 255). You can see the following example function in assembly, if that helps:
![image](https://github.com/Open-Agriculture/AgIsoStack-plus-plus/assets/29043784/a7d08321-6fc9-4d03-be1b-8a3b0afce34f)

If this was about a **less or equal** operator or something, then it would be another story; 
![image](https://github.com/Open-Agriculture/AgIsoStack-plus-plus/assets/29043784/c44f91af-a9fc-4315-bb15-39b7deb48de3)
here the loop can never reach it's end condition since no uint8_t can be higher than 255

Reply:
Ah ok